### DISCUSSION TITLE
getInputProps callbacks

### DISCUSSION THREAD

**Discussion by ghost** [mantinedev/mantine#1391] [Feature requests]
Hello!

It would be great that callback functionality would be added. For example, the form error automatically disappears when the value in the field changes, but no other functions can be assigned. 

I can now offer a temporary solution:

    function useCustomFormHelper<T>(form: UseFormReturnType<T>, onChangeCustomCallback: Function) {
    let tempGetInputPropsFn = form.getInputProps;
    type getInputPropsFnParameters = Parameters<typeof tempGetInputPropsFn>;
    type getInputPropsFnReturn = ReturnType<typeof tempGetInputPropsFn>;

    function customGetInputPropsFn(...parameters: getInputPropsFnParameters): getInputPropsFnReturn {
      let {...payload} = tempGetInputPropsFn(...parameters);
      let tempOnChange = payload.onChange;
      payload.onChange = function(val) {
        tempOnChange(val);
        onChangeCustomCallback();
      }
      return payload;
    }
    form.getInputProps = customGetInputPropsFn as typeof tempGetInputPropsFn;
    return form;
    }

**Comment by rtivital** [comment]
`getInputProps` is a shorthand for `onChange`, `value` and `error` props, you can set all these props individually without `getInputProps` using `form.setFieldValue`, `form.values` and `form.errors`.
